pkg/ingestor/verifier/sigstore_verifier/sigstore_verifier.go

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sigstore_verifier
+
+import (
+	"bytes"
+	"crypto"
+	"encoding/json"
+	"fmt"
+
+	"github.com/guacsec/guac/pkg/ingestor/key"
+	"github.com/guacsec/guac/pkg/ingestor/verifier"
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+	"github.com/sigstore/sigstore/pkg/signature"
+	sig_dsse "github.com/sigstore/sigstore/pkg/signature/dsse"
+)
+
+type SigstoreVerifier struct {
```

Comment:
should be private if exposing `NewSigstoreVerifier()`

Reply:
add TODO note that this only supports SHA256? (or is that the only one supported?)

Reply:
so that is the default one that supported by in-toto and rekor. I will add in the note.